lib/compute/agent-node-config.ts

```diff
@@ -107,11 +106,11 @@ export class AgentNodeConfig {
      );
 
      /* eslint-disable eqeqeq */
-     if (assumeRole.toString() !== 'undefined') {
+     if (assumeRole) {
```

Comment:
The `assumeRole` is not a boolean, have you tested this @gaiksaya ?

Reply:
Yes, works as expected. Also the tests cover this scenario here in this file `test/compute/agent-node-config.test.ts`

Reply:
Also if the resource string array checks for the validity of the resource we are adding, it cannot be blank or random values has to be a valid resource like arns or *